src/palindrom.js

```diff
@@ -418,39 +371,38 @@ var Palindrom = (function () {
    * @returns {XMLHttpRequest} performed XHR
    */
   PalindromNetworkChannel.prototype.xhr = function (url, accept, data, callback, setReferer) {
-    var that = this;
-    var req = new XMLHttpRequest();
-    var method = "GET";
-    req.onload = function () {
-      var res = this;
-      that.handleResponseHeader(res);
-      if (res.status >= 400 && res.status <= 599) {
-        that.onFatalError({ statusCode: res.status, statusText: res.statusText, reason: res.responseText }, url, method);
-      }
-      else {
-        callback && callback.call(that.palindrom, res, method);
-      }
-    };
-    req.onerror = that.onConnectionError.bind(that);
-    if (data) {
-      method = "PATCH";
-      req.open(method, url, true);
-      req.setRequestHeader('Content-Type', 'application/json-patch+json');
-    }
-    else {
-      req.open(method, url, true);
-    }
-    if (accept) {
-      req.setRequestHeader('Accept', accept);
-    }
-    if (that.remoteUrl && setReferer) {
-      req.setRequestHeader('X-Referer', that.remoteUrl.pathname);
-    }
-    that.onSend(data, url, method);
-    req.send(data);
 
-    return req;
-  };
+        var method = data ? 'PATCH' : 'GET';
+        var headers = {};
+
+        if (data) {
+            headers['Content-Type'] = 'application/json-patch+json';
+        }
+        if (accept) {
+            headers['Accept'] = accept;
+        }
+        if (this.remoteUrl && setReferer) {
+            headers['X-Referer'] = this.remoteUrl.pathname;
+        }
+        var successHandler = function (res) {
+            this.handleResponseHeader(res);
+            callback && callback.call(this.palindrom, res, method);
+        }.bind(this);
```

Comment:
What about providing reference to this explicitly in the closure, instead of redefining function with `bind`?
(same applies for `failureHandler`)